File: src/main/R/Slide.R
# This represents the base class for Slides

Slide <- setRefClass(
  Class = "Slide",
  fields = list(
    hidden = "logical",
    layout = "character"
  ),
  methods = list(
    
    isHidden = function() {
      hidden
    },
    
    setHidden = function(hidden) {
      "Set the hidden attribute of this Slide"
      hidden <<- as.logical(hidden)
      invisible(.self)
    },
    
    setLayout = function(layout) {
      "Set the layout of this Slide"
      layout <<- as.character(layout)
      invisible(.self)
    },
    
    toJSON = function() {
      paste0('{ "hidden" : "', hidden, '", "layout" : "', layout, '" }')
    }
  )
)

File: src/main/java/se/alipsa/presentations/Slide.java
package se.alipsa.presentations;

public class Slide {

  private boolean hidden = false;
  private String layout;

  public boolean isHidden() {
    return hidden;
  }

  public void setHidden(boolean hidden) {
    this.hidden = hidden;
  }

  public String getLayout() {
    return layout;
  }

  public void setLayout(String layout) {
    this.layout = layout;
  }
}